### PR TITLE
Async storage dev

### PR DESCRIPTION
**Pull request by Dileepnidi**
Added skeleton with  interface for RNS AsyncStorage

### CODE REVIEW

File: skia/RSAsyncStorage.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+#include <cxxreact/JsArgumentHelpers.h>
+
+#include "ReactSkia/utils/RnsLog.h"
+#include "RSAsyncStorage.h"
+using namespace folly;
+namespace facebook {
+namespace xplat {
+
+RSAsyncStorage::RSAsyncStorage(){
```
Comment (Dileepnidi) [inline skia/RSAsyncStorage.cpp]: Rename RSAsyncStorage as RSAsyncStorageModule

File: skia/BUILD.gn [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    "-Wno-sign-compare",
+    "-Wno-header-hygiene",
+  ]
+
+  include_dirs = [ ".",
+                  "//react-native/ReactCommon/",
+                  ]
+}
+
+rn_cxx_component("RnAsyncStorage") {
```
Comment (Dileepnidi) [inline skia/BUILD.gn]: rename RnAsyncStorage as RNSAsyncStorage

File: skia/RSAsyncStorage.cpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+auto RSAsyncStorage::getConstants() -> std::map<std::string, folly::dynamic> {
+  return {};
+}
+
+std::string RSAsyncStorage::getName() {
+  return "PlatformLocalStorage";
+}
+
+auto RSAsyncStorage::getMethods() -> std::vector<Method> {
+  RNS_LOG_ERROR("RSAsyncStorage::getMethods");
```
Comment (Dileepnidi) [inline skia/RSAsyncStorage.cpp]: Remove this line

**Review by munezbn — APPROVED** [review]

File: skia/BUILD.gn [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  sources = [
+    "RSAsyncStorageModule.h",
+    "RSAsyncStorageModule.cpp",
+  ]
+
+  public_configs = [ ":LibRNSAsyncStorage_config" ]
+  public_configs += [ "//build/secondary/folly:folly_config" ]
+  deps = [
+    "//folly",
+    "//skia",
```
Comment (pprabhu-ng) [inline skia/BUILD.gn:30]: why is skia required for this module ?

File: skia/BUILD.gn [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  include_dirs = [ ".",
+                  "//react-native/ReactCommon/",
+                  ]
+}
+
+rn_cxx_component("RNSAsyncStorage") {
+
+  # RNS Native module and its dependencies
+  sources = [
+    "RSAsyncStorageModule.h",
```
Comment (pprabhu-ng) [inline skia/BUILD.gn:22]: is the "RS" prefix proper for these files?

**Review by pprabhu-ng — APPROVED** [review]

File: skia/RSAsyncStorageModule.h [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+class RSAsyncStorageModule : public module::CxxModule {
+ public:
+  RSAsyncStorageModule();
+  virtual auto getConstants() -> std::map<std::string, folly::dynamic>;
+  virtual auto getMethods() -> std::vector<Method>;
+  std::string getName();
+};
+#ifdef __cplusplus
+extern "C" {
```
Comment (pprabhu-ng) [inline skia/RSAsyncStorageModule.h:25]: is extern "C" required for this module ?

**Pull request merged**